day-6/stefan-excel.py

```diff
@@ -1,47 +1,69 @@
-#!/usr/bin/env python
-
-# You'll need to pip import openpyxl
-import openpyxl
-
-excelFile = openpyxl.load_workbook('stefan-data.xlsx')
-
-sheet1 = excelFile.active
-
-for row in sheet1.iter_rows():
-    for cell in row:
-        if cell.value == 0:
-            cell.value = '<10'
-        elif cell.value == 1:
-            cell.value = '20'
-        elif cell.value == 2:
-            cell.value = '40'
-        elif cell.value == 3:
-            cell.value = '80'
-        elif cell.value == 4:
-            cell.value = '160'
-        elif cell.value == 5:
-            cell.value = '320'
-        elif cell.value == 6:
-            cell.value = '640'
-        elif cell.value == 7:
-            cell.value = '1280'
-        elif cell.value == 0.5:
-            cell.value = '10'
-        elif cell.value == 1.5:
-            cell.value = '30'
-        elif cell.value == 2.5:
-            cell.value = '60'
-        elif cell.value == 3.5:
-            cell.value = '120'
-        elif cell.value == 4.5:
-            cell.value = '240'
-        elif cell.value == 5.5:
-            cell.value = '480'
-        elif cell.value == 6.5:
-            cell.value = '960'
-        elif cell.value == 7.5:
-            cell.value = '1920'
-        elif cell.value == 8.5:
-            cell.value = '2560'
-
-excelFile.save('stefan-new-data.xlsx')
+#!/usr/bin/env python
+
+# You'll need to pip import openpyxl 
+import openpyxl
+import argparse
+
+parser = argparse.ArgumentParser( 
+    description='Script to convert HI data from lab readout to titers.',
+    usage='python3  stefan-excel.py -e [table.xlsx]')
+
+parser.add_argument(
+    '-e', action="store", default= None, required=True,
+    help='location of Excel file to parse')
+
+parser.add_argument(
+    '-s', type=int, default=20,
+    help='The starting dillution, default=20.')
+
+
+
+
+args = parser.parse_args()
+std = args.starting_dillution
+stdh = std * 1.5
+co = round(std // 2)
+cutoff = '<%s' % co
+
+table = openpyxl.load_workbook(args.Excelfile)
+
+sheet1 = table.active
+
+convert = {
+    '0' : cutoff,
+    '0.5' : co,
+    '1' : std,
+    '1.5' : stdh,
+    '2' : std*(2**1),
+    '2.5' : stdh*(2**1),
+    '3' : std*(2**2),
+    '3.5' : stdh*(2**2),
+    '4' : std*(2**3),
+    '4.5' : stdh*(2**3),
+    '5' : std*(2**4),
+    '5.5' : stdh*(2**4),
+    '6' : std*(2**5),
+    '6.5' : stdh*(2**5),
+    '7' : std*(2**6),
+    '7.5' : stdh*(2**6),
+    '8' : std*(2**7),
+    '8.5' : stdh*(2**7),
+    '9' : std*(2**8),
+    '9.5' : stdh*(2**8),
+    '10' : std*(2**9),
+    '10.5' : stdh*(2**9),
+    '11' : std*(2**10),
+    '11.5' : stdh*(2**10),
+    '12' : std*(2**11),
+
+}
+
+for row in sheet1.iter_rows():
+    for cell in row:
+        value = str(cell.value)
+        if cell.value in convert:
+            cell.value = convert.get(str(cell.value), cell.value)
+
+
+name, blank = args.Excelfile.split('.xlsx')
```

Comment:
It's a convention to use an underscore for variables you don't care about & won't use. E.g.,

```python
name, _ = args.Excelfile.split('.xlsx')
```

It's only a convention, but it's a widely used one. It tells the reader immediately that you're throwing away one value, and they don't then have this pending "what is `blank`?" thought in mind.

Or, do this:

```python
name = args.Excelfile.split('.xlsx')[0]
```

Reply:
thanks